aloe/registry.py

```diff
@@ -214,8 +214,12 @@ def load(self, sentence, func):
         self.steps[step_re.pattern] = (step_re, func)
 
         try:
+            if not getattr(func, 'patterns', None):
+                func.patterns = set()
+
+            func.patterns.add(step_re.pattern)
             func.sentence = sentence
-            func.unregister = partial(self.unload, step_re.pattern)
+            func.unregister = partial(self.unload, *func.patterns)
```

Comment:
I understand what this is doing. But where is the `patterns` attribute used, other than right here, where it can be replaced by a local variable?